halo2-base/src/gates/flex_gate.rs

```diff
@@ -272,6 +362,9 @@ pub trait GateInstructions<F: ScalarField> {
     where
         QA: Into<QuantumCell<F>>;
 
+    /// Constrains and returns the sum of [QuantumCell]'s in iterator `a` to the last element or [None] yielded by the iterator.
```

Comment:
I don't think None is possible?

"to the last element" is confusing, just say it returns the sum of `a`